### DISCUSSION TITLE
[* DateTimeV2] - how to use timex/otherways to determine type of dates it found (if quarter or not)

### DISCUSSION THREAD

**Discussion by zoinkydoink** [microsoft/Recognizers-Text#2501] [Q&A]
If i pass the string "This quarter", I get the proper dates and comes with `timex` `(2021-01-01,2021-04-01,P3M)`
which is great how can i  identify this was actually for a quarter? do I have to check the start and end date that it retrieved or is there a better way?

If i pass the string "First quarter of the year", then I get 2 sets of dates one for q1 in 2020 and one for q1 in 2020 but the` timex` are diff then above `(XXXX-01-01,XXXX-04-01,P3M)` is for 2020 set of dates and `(XXXX-01-01,XXXX-04-01,P3M)` for 2021 set of dates.

I guess my question is is there an easy way to use the `timex` or something else to identify what it found without having to compare dates etc. What is the purpose of `timex` and how/if we can use to  identify findings

I also have the EnablePrevew flag on

Also shouldnt both strings return one set of date and that is for this year instead of one of them returning two that includes last year?

**Comment by tellarin** [comment]
Reasoning over the extracted entity should take into account all attributes: type, timex, and modifiers.
In this case, the timex represents a date range, and the last term in it ("P3M") specifies it is a period of 3 months.

Regarding the resolution, the recognizers return both a past and a future resolution, if the entity mention is un-anchored.
"First quarter of 2021" is anchored in 2021; "First quarter of the year" is not, it depends on whatever date is passed as reference time.

Does this answer your question?

Please use 'Discussions' instead of 'Issues' in GitHub for such queries that are not bugs/issues/feature requests, ok? It helps with bug triage. Thanks.
